code/app/commodity/server.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/brahma-adshonor/microservice/code/app/commodity/model"
+	"gorm.io/driver/mysql"
+	"gorm.io/gorm"
+)
+
+var (
+	db  *gorm.DB
+	err error
+)
+
+func hello(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, "hello\n")
+}
+func getProduct(w http.ResponseWriter, r *http.Request) {
+	var request ProductRequst
+	var response ProductResponse
+
+	fmt.Printf("http body is : %v\n", r.Body)
+	err := json.NewDecoder(r.Body).Decode(&request)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	fmt.Printf("ProductRequest is : %v\n", request)
+
+	var result *gorm.DB
+	if request.All {
+		result = db.Find(&response.Products)
```

Comment:
should with a limitation or pagination?

Reply:
as mentioned above, the security related could be add in next few chapters